Cython/Utility/StringTools.c

```diff
@@ -118,25 +118,36 @@ static CYTHON_INLINE int __Pyx_UnicodeContainsUCS4(PyObject* unicode, Py_UCS4 ch
 
 //////////////////// PyUCS4InUnicode ////////////////////
 
-#if PY_VERSION_HEX < 0x03090000
-#if Py_UNICODE_SIZE == 2
-static int __Pyx_PyUnicodeBufferContainsUCS4_SP(Py_UNICODE* buffer, Py_ssize_t length, Py_UCS4 character) {
+#if PY_VERSION_HEX < 0x03090000 || (defined(PyUnicode_WCHAR_KIND) && defined(PyUnicode_AS_UNICODE))
+
+#ifdef PyUnicode_AS_UNICODE
+#define __Pyx_UNICODE Py_UNICODE
+#define __Pyx_PyUnicode_AS_UNICODE(op) PyUnicode_AS_UNICODE(op)
+#define __Pyx_PyUnicode_GET_SIZE(op) PyUnicode_GET_SIZE(op)
+#else
+#define __Pyx_UNICODE wchar_t
+#define __Pyx_PyUnicode_AS_UNICODE(op) (((PyASCIIObject *)(op))->wstr)
```

Comment:
If I understand PEP623 correctly `wstr` and `PyUnicode_WSTR_LENGTH` are removed at the same time as `PyUnicode_AS_UNICODE` so I'm not sure they're useful fallbacks.

Reply:
Erm, right. I think the `#ifdef` above is wrong. The fallback should probably just always be used in Py3.9+ to avoid deprecated API calls, until `PyUnicode_WCHAR_KIND` goes out of existence (which is handled further up).